snapcraft/internal/parts.py

```diff
@@ -117,6 +124,22 @@ def get_part(self, part_name, full=False):
                 remote_part.pop(key)
         return remote_part
 
+    def matches_for(self, part_match, max_len=0):
+        matcher = difflib.SequenceMatcher(isjunk=None, autojunk=False)
+        matcher.set_seq2(part_match)
+
+        matching_parts = {}
+        for part_name in self._parts.keys():
+            matcher.set_seq1(part_name)
+            add_part_name = matcher.ratio() >= _MATCH_RATIO
+
+            if add_part_name or part_match in part_name:
```

Comment:
Use parentheses here to make the rules of precedence clear:
if add_part_name or (part_match in part_name):